amr-wind/wind_energy/ABLFieldInit.cpp

```diff
@@ -193,13 +230,52 @@ void ABLFieldInit::operator()(
                     }
                 }
 
+                temperature(i, j, k, 0) += theta;
+                velocity(i, j, k, 0) += umean_prof;
+                velocity(i, j, k, 1) += vmean_prof;
+            });
+    } else if (m_initial_wind_profile) {
+        //! RANS 1-D profile
+        const amrex::Real* windh = m_windht_d.data();
+        const amrex::Real* uu = m_prof_u_d.data();
+        const amrex::Real* vv = m_prof_v_d.data();
+
+        amrex::ParallelFor(
+            vbx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+
+                density(i, j, k) = rho_init;
+                amrex::Real theta = tv[0];
+                amrex::Real umean_prof = uu[0];
+                amrex::Real vmean_prof = vv[0];
+
+                for (int iz = 0; iz < ntvals - 1; ++iz) {
```

Comment:
Can you use the bisection search for these interpolations too?

Reply:
This was an existing code before my PR and I did not want to make any changes since I was not sure why the method was chosen.

Reply:
I would still replace these using `interp::linear` cleaner code and easier to maintain in the future. You don't necessarily need the double call of `bisection_search + linear_impl` for these.